src/backend/distributed/utils/node_metadata.c

```diff
@@ -1424,7 +1424,10 @@ TupleToWorkerNode(TupleDesc tupleDescriptor, HeapTuple heapTuple)
 	{
 		Name nodeClusterName = DatumGetName(nodeCluster);
 		char *nodeClusterString = NameStr(*nodeClusterName);
-		strlcpy(workerNode->nodeCluster, nodeClusterString, NAMEDATALEN);
+		if (nodeClusterString != NULL)
```

Comment:
I found the reason for this. We call initialization function before we add nodecluster column.

```sql
SELECT master_initialize_node_metadata();
```

before

```sql
ALTER TABLE pg_dist_node ADD COLUMN nodecluster name NOT NULL DEFAULT 'default';
```
@aamederen, I guess the only thing we can do is to add a comment to your null check and explain the reason.